schema.py

```diff
@@ -126,21 +123,10 @@ def validate(self, data):
                     except SchemaError:
                         pass
                     else:
-                        try:
-                            nvalue = Schema(svalue, error=e).validate(value)
-                        except SchemaError as _x:
-                            x = _x
-                            raise
-                        else:
-                            coverage.add(skey)
-                            valid = True
-                            break
-                if valid:
-                    new[nkey] = nvalue
-                elif skey is not None:
-                    if x is not None:
```

Comment:
`x` is always `None` at this point:
* It has been initialized as `None`.
* When a new value is assigned to `x` in line 132, it is followed directly by `raise` with the effect that this if-statement is never reached.